### DISCUSSION TITLE
Dark edges when compose files with transparency

### DISCUSSION THREAD

**Discussion by M-Sadeea** [ImageMagick/ImageMagick#5324] [Help] [answered]
Hi, I used this command to mask image.png according to map.png:
`convert map.png image.png -compose src-in -composite out.png` 
but unfortunately there was a dark edges around the result .. how can I prevent this?
thanks.

![image](https://user-images.githubusercontent.com/18248687/178761377-7d1c1021-b9b6-44ef-8bd9-769e731f5fb9.png)

![map](https://user-images.githubusercontent.com/18248687/178761396-40b1bdba-e554-48ac-ab99-97eeffe68bb8.png)
map.png .. it is a white circle!

![out](https://user-images.githubusercontent.com/18248687/178761398-e5162b0f-d6bb-4470-9fff-fdb295ea2ae2.png)

I am using ImageMagick 7.1.0-39 on Mac-OS-10.15

**Comment by fmw42** [comment, marked as answer]
This should work. I am using IM 7.1.0-43 Q16-HDRI on M1 Mac mini with Monterey.

`magick plant.png \( circle.png -alpha extract \) -compose copy_opacity -composite result.png`

or

`magick plant.png circle.png +swap -compose in -composite result.png`

![result](https://user-images.githubusercontent.com/13458218/178765335-c19ccc24-15cf-4bab-b114-5e2b5b9707c5.png)

**Reply by M-Sadeea** [reply]
Thank you very much @fmw42 for your fast replay, the first command works, but not the second .. 
I spent hours searching for a solution .. thank a lot :)

**Reply by fmw42** [reply]
Perhaps your version of IM 7 has a bug. Try upgrading. Also be sure to use the latest delegates.
